clients/src/main/java/org/apache/kafka/common/requests/FetchResponse.java

```diff
@@ -114,8 +143,28 @@ public FetchResponse(Struct struct) {
                 int partition = partitionResponseHeader.getInt(PARTITION_KEY_NAME);
                 Errors error = Errors.forCode(partitionResponseHeader.getShort(ERROR_CODE_KEY_NAME));
                 long highWatermark = partitionResponseHeader.getLong(HIGH_WATERMARK_KEY_NAME);
+                long lastStableOffset = INVALID_LSO;
+                if (partitionResponse.hasField(LAST_STABLE_OFFSET_KEY_NAME))
```

Comment:
Bug: we should use `partitionResponseHeader` instead of `partitionResponse` to get the response. I guess this is not discovered by any test because the current patch always have `last_stable_offset = -1L`?